### ISSUE TITLE
DNR (set_undefibbable) can be triggered in situations where it shouldn't be

### ISSUE THREAD

**Issue by DeltaFire15** [tgstation/TerraGov-Marine-Corps#14745]
### Description

After hearing about some questionable cases of DNRs triggering, I did some investigating and found that #14536 , while fixing some minimap things and situations where people did not go DNR, also broke some things that **make you unrevivable despite the fact you shouldn't be.**

This is done by this section of the code
![image](https://github.com/tgstation/TerraGov-Marine-Corps/assets/46569814/0e2041b2-a591-409c-b993-f43a18c8c3ef)

The issue here is that there are two cases where this logic will return "true" where it shouldn't:
If you are a species that can be decapped & not become unrevivable (Robots and Synthetics), it will be triggered on decapitation, as your mind will go to the head (!client = TRUE) and will have you not be a ghost of the body (because you got decapped so your "return to body" will go to the head itself / the ghost is associated with it) so the !related_ghost is also TRUE.

Another case is that if you are in your body and crash (or restart your game) after dying, you will also be marked as DNR, as you are not there (not connected, !client = TRUE) and aren't ghosted so the ghost search doesn't turn up anything either.

These two are cases where you should be revivable (either via the associated surgery if decapped / simply via reconnecting if DCd, but instead will be marked DNR by the game).

tl;dr; if you get decapped as synth / robot or you disconnect / crash while dead & in your body, the game will mark you DNR.

### Test Merges

Irrelevant.

### Reproduction Steps

1. Get decapped as Synth / Robot.
2. Get your head reattached.
3. Get defibbed.
4. Get a DNR / Body decayed message.

Alt:
1. Die.
2. Be in your body.
3. Disconnect / Crash.
4. Reconnect.
5. Be marked DNR anyways.

### Screenshots

Should be sufficiently explained.
